uvm/impl/com/untangle/uvm/SettingsManagerImpl.java

```diff
@@ -306,6 +307,36 @@ public void save( String fileName, String inputFilename, boolean saveVersion ) t
         _saveImpl( fileName, inputFilename, saveVersion, true );
     }
 
+     /**
+     *Retrieve  username and hostname for reports app
+     *  @param  appName for which data is fetched
+     *
+     *  @return A string with comma separated username and hostname
+     */
+    public String getUserAndHostNameInfo(String appName) {
+
+        String username = null;
+        String hostname = null;
+        if((UvmContextImpl.getInstance().threadRequest() != null) &&
```

Comment:
May be `UvmContextImpl.getInstance().threadRequest()` can be extracted into a variable so that it can be reused in all other occurrences